packages/common/src/sandbox/modules.ts

```diff
@@ -251,14 +252,30 @@ export const isMainModule = (
 export const findMainModule = (
   modules: Module[],
   directories: Directory[],
-  entry: string = 'index.js'
+  entry: string = 'index.js',
+  template?: TemplateType
 ) => {
   try {
+    // first attempt: try loading the entry file if it exists
     const module = resolveModule(entry, modules, directories);
-
     return module;
   } catch (e) {
-    return modules[0];
+    try {
+      // second attempt: try loading the first file that exists from
+      // the list of possible defaults in the template defination
+      const templateDefinition = getTemplateDefinition(template);
+
+      const defaultOpenedFiles = templateDefinition.getDefaultOpenedFiles({});
```

Comment:
Ah, normally we pass in the parsed configurations (like `package.json`) here, in case it's needed to determine the main file. We use the function `parseSandboxConfigurations` (https://github.com/CompuIves/codesandbox-client/blob/b228282987eb687bf4251c1fb564ec6467e32d6c/packages/app/src/app/store/actions.js#L173) to get these configurations, but that function needs `sandbox`, which is not available here.

It would be good to use the parsed configurations as well, to keep consistency with the editor, but we'd probably need to get another argument here with parsed configurations...

Reply:
I couldn't import that exact file because of a cyclic dependency. But was able to repurpose it's implementation.